src/util-logopenfile.c

```diff
@@ -209,9 +209,19 @@ static int SCLogFileWriteNoLock(const char *buffer, int buffer_len, LogFileCtx *
     }
 
     if (log_ctx->fp) {
-        clearerr(log_ctx->fp);
-        ret = SCFwriteUnlocked(buffer, buffer_len, 1, log_ctx->fp);
-        fflush(log_ctx->fp);
+        SCClearErrUnlocked(log_ctx->fp);
+        if (1 != SCFwriteUnlocked(buffer, buffer_len, 1, log_ctx->fp)) {
+            if (SCFerrorUnlocked(log_ctx->fp)) {
```

Comment:
what happens if we do not get an error from `SCFerrorUnlocked` here?

I wonder if we should skip this check and in the error output do something like:
```c
SCFerrorUnlocked(log_ctx->fp) ? strerror(errno) : "unknown error"
```

Reply:
Make sense ... I doubt it'll happen but the overhead is minimal.